invenio_stats/templates.py

```diff
@@ -21,3 +23,17 @@ def register_templates():
     ]
 
     return event_templates + aggregation_templates
+
+
+def register_templates():
+    """Register search templates for events."""
+    if current_app.config["STATS_REGISTER_INDEX_TEMPLATES"]:
+        return []
```

Comment:
I'm not sure if it makes sense to show a deprecation warning without having a "way out". Usually these warnings link to some documentation that explains why the change is happening and how to migrate to the new solution (see example below).

I think to move things forward it would be better to cut a minor release, and include this deprecation warning in a new PR that would also include this extra information in the text (and release as a minor/patch).

---

**GH Actions deprecation warning**

<img width="780" alt="image" src="https://github.com/user-attachments/assets/7d865e4e-73cf-4774-abb6-2ed960859e7f">

**SQLAlchemy**

```
SADeprecationWarning: When transforming <class '__main__.User'> to a
dataclass, attribute(s) "create_user", "update_user" originates from
superclass <class
'__main__.Mixin'>, which is not a dataclass. This usage is deprecated and
will raise an error in SQLAlchemy 2.1. When declaring SQLAlchemy
Declarative Dataclasses, ensure that all mixin classes and other
superclasses which include attributes are also a subclass of
MappedAsDataclass.
```